core/sql/cli/memorymonitor.cpp

```diff
@@ -223,12 +229,12 @@ void MemoryMonitor::update(float &scale) {
 	char * currPtr;
         bytesRead = fread(buffer, 1, 2048, fd_meminfo_);
         // Make sure there wasn't an error (next fseek will clear eof)
-        if (!feof(fd_meminfo_))// Make sure there wasn't an error
-        {
-	        scale = 6;
-		pressure_ = 0;
-		return;
-        }
+        if (ferror(fd_meminfo_))
+           assert(false); 
+        if (feof(fd_meminfo_))
+           clearerr(fd_meminfo_); 
+        else
+           buffer[bytesRead] = '\0';
```

Comment:
If we get an EOF, which should be the normal case, then shouldn't we also set buffer[bytesRead] to \0? In either case, to do that, we would need to allocate one extra byte to avoid a buffer overrun.